caesar_cipher.rb

```diff
@@ -1,25 +1,28 @@
+$lowercase_letters = ('a'..'z').to_a
+$uppercase_letters = ('A'..'Z').to_a
+
+def get_shifted_idx(style, char, shift)
+  if style == 'lowercase'
+    shifted_idx = $lowercase_letters.index(char) + shift
+  else
+    shifted_idx = $uppercase_letters.index(char) + shift
+  end
+  if shifted_idx > $lowercase_letters.size
+    shifted_idx -= $lowercase_letters.size
+  end
+  return shifted_idx
+end
+
 def caesar_cipher(str, shift) 
-    lowercase_letters = Array('a'..'z')
-    uppercase_letters = Array('A'..'Z')
-    new_str = ''
-    str.each_char do |c|
-        if lowercase_letters.include?(c)
-            shifted_idx = lowercase_letters.index(c) + shift
-            # if the new idx is out of bounds, loop back to beginning for idx
-            # take idx value and subtract from 26 to get new idx value
-            if shifted_idx > 26
-                shifted_idx -= 26
-            end
-            new_str += lowercase_letters[shifted_idx]
-        elsif uppercase_letters.include?(c)
-            shifted_idx = uppercase_letters.index(c) + shift
-            if shifted_idx > 26
-                shifted_idx -= 26
-            end
-            new_str += uppercase_letters[shifted_idx]       
-        else
-            new_str += c       
-        end
+  new_str = ''
+  str.each_char do |c|
+    if $lowercase_letters.include?(c)
+      new_str += $lowercase_letters[get_shifted_idx('lowercase', c, shift)]
```

Comment:
How about rather than passing in `'lowercase'` or `'uppercase'`, you just passed the array directly? I think that might allow you to get rid of your `if/else` in the `get_shifted_idx` method?

Reply:
wow!! agreed! thanks for the suggestion